KlaviyoV3Sdk/KlaviyoV3Api.php

```diff
@@ -0,0 +1,490 @@
+    }
+
+    /**
+     * Build headers for the Klaviyo all event
+     *
+     * @param $clientEvent
+     * @return array|array[]
+     */
+    public function getHeaders()
+    {
+        $klVersion = $this->_klaviyoScopeSetting->getVersion();
+
+        $objectManager = \Magento\Framework\App\ObjectManager::getInstance();
+        $productMetadata = $objectManager->get('Magento\Framework\App\ProductMetadataInterface');
+        $m2Version = $productMetadata->getVersion();
+
+        $headers = array(
+            CURLOPT_HTTPHEADER => [
+                self::REVISION_KEY_HEADER . ': ' . self::KLAVIYO_V3_REVISION,
+                self::CONTENT_TYPE_KEY_HEADER . ': ' . self::APPLICATION_JSON_HEADER_VALUE,
+                self::ACCEPT_KEY_HEADER . ': ' . self::APPLICATION_JSON_HEADER_VALUE,
+                self::KLAVIYO_USER_AGENT_KEY . ': ' . 'magento2-klaviyo/' . $klVersion . ' Magento2/' . $m2Version . ' PHP/' . phpversion(),
+                self::AUTHORIZATION_KEY_HEADER . ': ' . self::KLAVIYO_API_KEY . ' ' . $this->private_key
+            ]
+        );
+
+        return $headers;
+    }
+
+
+    /**
+     * Query for all available lists in Klaviyo
+     * https://developers.klaviyo.com/en/v2023-08-15/reference/get_lists
+     *
+     * @return array
+     * @throws KlaviyoApiException
+     * @throws KlaviyoAuthenticationException
+     * @throws KlaviyoRateLimitException
+     */
+    public function getLists()
+    {
+        $response_body = $this->requestV3('api/lists/', self::HTTP_GET);
+
+        return $response_body[self::DATA_KEY_PAYLOAD];
+    }
+
+    /**
+     * Search for profile by Email
+     * https://developers.klaviyo.com/en/v2023-08-15/reference/get_profiles
+     *
+     * @param $email
+     * @return false|mixed
+     */
+    public function searchProfileByEmail($email)
+    {
+        $response_body = $this->requestV3("api/profiles/?filter=equals(email,'$email')", self::HTTP_GET);
+
+        if (empty($response_body[self::DATA_KEY_PAYLOAD])) {
+            return false;
+        } else {
+            $id = $response_body[self::DATA_KEY_PAYLOAD][0][self::ID_KEY_PAYLOAD];
+            return [
+                'response' => $response_body,
+                'profile_id' => $id
+            ];
+        }
+    }
+
+    /**
+     * Add a Profile to a list using profile id
+     * https://developers.klaviyo.com/en/v2023-08-15/reference/create_list_relationships
+     *
+     * @param $list_id
+     * @param $profile_id
+     * @return array|string|null
+     * @throws KlaviyoApiException
+     */
+    public function addProfileToList($list_id, $profile_id)
+    {
+        $body = array(
+            self::DATA_KEY_PAYLOAD => array(
+                self::TYPE_KEY_PAYLOAD => self::PROFILE_KEY_PAYLOAD,
+                self::ID_KEY_PAYLOAD => $profile_id
+            )
+        );
+
+        return $this->requestV3("api/lists/$list_id/relationships/profiles/", self::HTTP_POST, $body);
```

Comment:
Need error handling for when this V3 API call fails. Or if the purpose here is to just send back the $response_body from the API call then the code that's making the addProfileToList() method call needs to handle the condition where the response is unexpected.

Reply:
Same comment for all the other places you are making the V3 API calls. We want to make sure the failure/unexpected cases are handled.

Reply:
We don't need the response from this method. This should only make the call to add the profile